src/OrbitDB.js

```diff
@@ -137,20 +148,20 @@ class OrbitDB {
   }
 
   async disconnect () {
-    // close Keystore
-    await this.keystore.close()
-
-    // close Cache
-    await Promise.all(Object.values(this.caches).map((cache) => {
-      return cache.close()
-    }))
+    // close keystores
+    const keystores = Object.values(this.keystores)
+    for (let keystore of keystores) {
+      await keystore.close()
+    }
+    this.keystores = []
 
     // Close all open databases
     const databases = Object.values(this.stores)
     for (let db of databases) {
       await db.close()
       delete this.stores[db.address.toString()]
     }
+    await this.caches['default'].close()
```

Comment:
Why only close the default cache and not others?

Reply:
The idea here is that the for loop above will call `db.close()` on each store, which will close any "custom" caches created via the `options.directory`. However, since writing this code I implemented a `this.cache` getter in `OrbitDB`, so I'll use that.